multicast/exceptions.py

```diff
@@ -0,0 +1,349 @@
+
+		>>> import multicast
+		>>>
+
+	Testcase 1: Recv should be automatically imported.
+
+		>>> multicast.exceptions.__name__ is not None
+		True
+		>>>
+
+"""
+
+
+try:
+	from . import sys  # skipcq: PYL-C0414
+	from . import argparse  # skipcq: PYL-C0414
+except Exception as err:
+	baton = ImportError(err, str("[CWE-758] Module failed completely."))
+	baton.module = __module__
+	baton.path = __file__
+	baton.__cause__ = err
+	raise baton from err
+
+
+class CommandExecutionError(RuntimeError):
+	"""
+	Exception raised when a command execution fails.
+
+	Attributes:
+		message (str) -- Description of the error.
+		exit_code (int) -- The exit code associated with the error.
+
+	Meta-Testing:
+
+		Testcase 1: Initialization with message and exit code.
+			A. - Initializes the error.
+			B. - checks inheritance.
+			C. - checks each attribute.
+
+			>>> error = CommandExecutionError("Failed to execute command", 1)
+			>>> isinstance(error, RuntimeError)
+			True
+			>>> error.message
+			'Failed to execute command'
+			>>> error.exit_code
+			1
+	"""
+
+	def __init__(self, *args, **kwargs):
+		"""
+		Initialize CommandExecutionError with a message and exit code.
+
+		Parameters:
+			message (str) -- Description of the error.
+			exit_code (int) -- The exit code associated with the error.
+			*args: Variable length argument list.
+			**kwargs: Arbitrary keyword arguments.
+
+		Raises:
+			TypeError: if the exit_code is not an int.
+
+		Meta-Testing:
+
+			Testcase 1: Initialization with different exit code:
+				A. - Initializes a CommandExecutionError with a specific exit code.
+				B. - Checks the message is still set, as super class would.
+				C. - check the specific exit code is 2.
+
+				>>> error = CommandExecutionError("Error message", 2)
+				>>> error.message
+				'Error message'
+				>>> error.exit_code
+				2
+		"""
+		if len(args) > 0 and isinstance(args[-1], int):
+			exit_code = args[-1]
+			args = args[:-1]
+		else:
+			exit_code = kwargs.pop("exit_code", 1)
+		super().__init__(*args, **kwargs)
+		self.message = args[0] if args else kwargs.get("message", "An error occurred")
+		self.exit_code = exit_code
+
+
+EXIT_CODES = {
+	0: (None, 'Success'),
+	1: (RuntimeError, 'General Error'),
+	2: (OSError, 'Misuse of Shell Builtins'),
+	64: (argparse.ArgumentError, 'Usage Error'),
+	65: (ValueError, 'Data Error'),
+	66: (FileNotFoundError, 'No Input'),
+	69: (ConnectionError, 'Unavailable Service'),
+	70: (Exception, 'Internal Software Error'),
+	77: (PermissionError, 'Permission Denied'),
+	125: (BaseException, 'Critical Failure'),
+	126: (AssertionError, 'Command Invoked Cannot Execute'),
+	127: (ModuleNotFoundError, 'Command Not Found'),
+	129: (None, 'Hangup (SIGHUP)'),
+	130: (KeyboardInterrupt, 'Interrupt (SIGINT)'),
+	134: (None, 'Abort (SIGABRT)'),
+	137: (None, 'Killed (SIGKILL)'),
+	141: (BrokenPipeError, 'Broken Pipe (SIGPIPE)'),
+	143: (SystemExit, 'Terminated (SIGTERM)'),
+	255: (None, 'Exit Status Out of Range'),
+}
```

Comment:
@coderabbitai, please read the file multicast/exceptions.py and the GH issue #117. Then please, generate a CEP-7 compliant docstring for EXIT_CODES. Be sure to include an extended description section and mention of CEP-8. Thanks in advance.

Reply:
```suggestion
EXIT_CODES = {
	0: (None, 'Success'),
	1: (RuntimeError, 'General Error'),
	2: (OSError, 'Misuse of Shell Builtins'),
	64: (argparse.ArgumentError, 'Usage Error'),
	65: (ValueError, 'Data Error'),
	66: (FileNotFoundError, 'No Input'),
	69: (ConnectionError, 'Unavailable Service'),
	70: (Exception, 'Internal Software Error'),
	77: (PermissionError, 'Permission Denied'),
	125: (BaseException, 'Critical Failure'),
	126: (AssertionError, 'Command Invoked Cannot Execute'),
	127: (ModuleNotFoundError, 'Command Not Found'),
	129: (None, 'Hangup (SIGHUP)'),
	130: (KeyboardInterrupt, 'Interrupt (SIGINT)'),
	134: (None, 'Abort (SIGABRT)'),
	137: (None, 'Killed (SIGKILL)'),
	141: (BrokenPipeError, 'Broken Pipe (SIGPIPE)'),
	143: (SystemExit, 'Terminated (SIGTERM)'),
	255: (None, 'Exit Status Out of Range'),
}
"""
Provides a mapping between exit codes and their corresponding exception classes and messages.

The `EXIT_CODES` dictionary serves as a centralized mapping for standard exit codes used within
the multicast module. Each key represents an exit code (as an integer), and each value is a tuple
containing the associated exception class (or `None` if not applicable) and a human-readable description of the exit condition.

CEP-8 Compliance:
	In accordance with [CEP-8]
	guidelines, this mapping facilitates consistent error handling and exit code management
	throughout the module. By associating specific exceptions with standard exit codes, the
	application adheres to predictable behavior in response to various error conditions, enhancing
	maintainability and debugging efficiency.

	Specific codes are detailed more in CEP-8.

Usage Example:
	```python
		from multicast.exceptions import EXIT_CODES
		from multicast.exceptions import get_exit_code_from_exception

		try:
			# Code that may raise an exception
			pass
		except Exception as e:
			exit_code = get_exit_code_from_exception(e)
			sys.exit(exit_code)
	```

Testing:

	Testcase 0: EXIT_CODES should be automatically imported.

		>>> import multicast
		>>> import multicast.exceptions
		>>> multicast.EXIT_CODES != None
		True
		>>> isInstance(multicast.EXIT_CODES, dict)
		True

"""
```